### PR TITLE
Update to Swift 3.0.2; add Jazzy docs

### PR DESCRIPTION
**Pull request by youming-lin**
Add more comments for Jazzy docs as well

### CODE REVIEW

File: .jazzy.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+theme: fullwidth
+clean: true
+exclude: [Packages]
+
+readme: README.md
+
+skip_undocumented: false
+hide_documentation_coverage: false
+
+xcodebuild_arguments: [-project, HTMLEntities.xcodeproj, -target, HTMLEntities, LIBRARY_SEARCH_PATHS=.build/debug]
```
Comment (ianpartridge) [inline .jazzy.yaml]: We don't need `LIBRARY_SEARCH_PATHS=.build/debug` any more. This should just be:

`xcodebuild_arguments: [-project, HTMLEntities.xcodeproj, -target, HTMLEntities]`

I think.

**Comment by youming-lin** [comment]
@ianpartridge I have removed the search paths from build arguments

**Comment by youming-lin** [comment]
@ianpartridge Thanks for the codecov run. Spotted some unused code that I'll remove in a new commit.

**Comment by youming-lin** [comment]
@ianpartridge I have removed ununused code and code coverage has improved to 90.75%. Can you review again and merge?

**Pull request merged**